### PR TITLE
upgrade: handle failures on save (bsc#1018110)

### PR DESCRIPTION
**Pull request by MaximilianMeister**
raise a custom exception when save fails, otherwise we might continue
silently

### CODE REVIEW

**Comment by MaximilianMeister** [comment]
[(2) S53P2 bsc#1018110 - Upgrade status library save() failures not handled](https://trello.com/c/S2piNMx2/245-2-s53p2-bsc-1018110-upgrade-status-library-save-failures-not-handled)

File: crowbar_framework/lib/crowbar/error/upgrade_status.rb [hunk 1]
```diff
@@ -41,5 +41,11 @@ def initialize(step_name = "")
         super("Step #{step_name} cannot be finished, as it is not running")
       end
     end
+
+    class SaveUpgradeStatusError < StandardError
+      def initialize(msg = "")
```
Comment (Itxaka) [inline crowbar_framework/lib/crowbar/error/upgrade_status.rb]: why give the value an empty default? Isnt the value always necessary?

Comment (MaximilianMeister) [inline crowbar_framework/lib/crowbar/error/upgrade_status.rb]: wow you're quick :) you're right i'll change it

**Comment by jsuchome** [comment]
Could you also adapt the clients?

File: crowbar_framework/lib/crowbar/error/upgrade_status.rb [hunk 2]
```diff
@@ -41,5 +41,11 @@ def initialize(step_name = "")
         super("Step #{step_name} cannot be finished, as it is not running")
       end
     end
+
+    class SaveUpgradeStatusError < StandardError
+      def initialize(msg)
+        super("Exception during saving the status file: #{msg}")
```
Comment (jsuchome) [inline crowbar_framework/lib/crowbar/error/upgrade_status.rb]: Sounds kind of strange. What about *Exception when saving...* ?

Comment (MaximilianMeister) [inline crowbar_framework/lib/crowbar/error/upgrade_status.rb]: hrm, i don't find it that strange. i just took it as it is. btw you introduced it here https://github.com/crowbar/crowbar-core/commit/814b2372120a4237bf1cbb3040230f5fb574fff4 ;-)

Comment (jsuchome) [inline crowbar_framework/lib/crowbar/error/upgrade_status.rb]: Yes, that looks like me. If you think it's correct English, just keep it.

**Pull request merged**